bionic/protocols.py

```diff
@@ -197,6 +198,39 @@ def __repr__(self):
         return f"{self.__class__.__name__}(...)"
 
 
+class JsonProtocol(BaseProtocol):
+    """
+    Decorator indicating that an entity's values can be serialized to the
+    ``JSON`` format.
+    """
+
+    def get_fixed_file_extension(self):
+        return "json"
+
+    def validate(self, value):
```

Comment:
So, while reading the docs for `json.dump`, I realized there's another edge case: a list or dict that contains itself. I've never seen a use for this, but it's actually well-behaved in Python: you can print it, pickle it, etc.

I think your code will still work here (by throwing a recursion exception, which will cause validation to fail), but we might want a test case for this.

Reply:
TIL you can pickle circular references. Added a test for it and it fails with `RecursionError`. I don't really like relying on this error but it still feels better than tracking all the seen objects during validation. Wdyt?

Reply:
Yeah, I could go either way. Relying on this particular error is not great, but it's also a super rare case.